onnx/shape_inference/implementation.cc

```diff
@@ -183,8 +183,34 @@ static void InferShapesImpl(
   }
 
   std::unordered_map<std::string, const TensorProto*> inputDataByName;
+  GraphProto inputGraph; // to not modify original graph; use temp graph  
   for (const auto& tp : g->initializer()) {
     inputDataByName[tp.name()] = &tp;
+    
+    // Support new IR: some tensors can only exist in initializer and not in input
+    // So shape_inference should make use of initializer shapes
+    // Store initializer shape info in value_info as well
+    TypeProto_Tensor* newTensorFromInitializer;
+    auto iter = valueTypesByName.find(tp.name());
+    // If it already exists in input, simply use the original input
+    if (iter != valueTypesByName.end()) {
+      continue; // skip if shape info exists in input 
+      // newTensorFromInitializer = valueTypesByName[tp.name()]->mutable_tensor_type();
+    }
+    // If not, add a new input into temp graph
+    else {
+      ValueInfoProto *newValueForInitializer = inputGraph.add_input();
+      newValueForInitializer->set_name(tp.name());
+      valueTypesByName[tp.name()]= newValueForInitializer->mutable_type();
+      newTensorFromInitializer = valueTypesByName[tp.name()]->mutable_tensor_type();
```

Comment:
I think "initializerTensorType" would be a better name for "newTensorFromInitializer" ... it is a tensor-type, not a tensor.